src/include/rimage/cavs/cavs_ext_manifest.h

```diff
@@ -0,0 +1,249 @@
+		uint32_t channel_7_0 : 1;           // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER
+		uint32_t channel_7_0_surround : 1;  // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT
+		uint32_t channel_7_1 : 1;           // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER
+		uint32_t channel_7_1_surround : 1;  // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT
+		uint32_t reserved : 16;
+	} r;
+};
+
+enum mod_stream_type
+{
+	ePcm = 0,                   ///< PCM stream
+	eMp3,                       ///< MP3 encoded stream
+	eAac,                       ///< AAC encoded stream
+	eMaxStreamType,
+	eStreamTypeInvalid = 0xFF
+};
+
+enum mod_type
+{
+	eBaseFwModule = 0,
+	eMixInModule,
+	eMixOutModule,
+	eCopierModule,
+	ePeakVolModule,
+	eUpDwMixModule,
+	eMuxModule,
+	eSrcModule,
+	eWovModule,
+	eFxModule,
+	eAecModule,
+	eKpbModule,
+	eMicSelectModule,
+	eFxFModule, //i.e.SmartAmp
+	eAudClassModule,
+	eFakeCopierModule,
+	eIoDriver,
+	eWhmModule,
+	eGdbStubModule,
+	eSensingModule,
+	eMaxModule,
+	eModuleInvalid = eMaxModule
+} ;
+
+// Note: pins description order is enough to perform pin association
+struct fw_pin_description
+{
+	union mod_pin_caps         caps;
+	enum mod_stream_type              format_type;                        // Supported Stream Type
+	union mod_sample_rates             sample_rate;       // Supported Sample Rate
+	union mod_sample_sizes             sample_size;   // Supported Valid Bits per sample
+	union mod_sample_containers        sample_container;    // Supported Bits per sample �container�
+	union mod_channel_config   ch_cfg;          // Supported channel configuration
+};
+
+struct fw_ext_man_header
+{
+	uint32_t    id;
+	uint32_t    len;           // sizeof(Extend Manifest) in bytes
+	uint16_t    version_major; // Version of Extended Manifest structure
+	uint16_t    version_minor; // Version of Extended Manifest structure
+	uint32_t    num_module_entries;
+};
+
+struct fw_ext_mod_config_header
+{
+	uint32_t            ext_module_config_length;       // sizeof(fw_ext_mod_config_header) in bytes
+	uint32_t            guid[4];                        // Module GUID
+	uint16_t            version_major;                  // Module version
+	uint16_t            version_minor;                  // Module version
+	uint16_t            version_hotfix;                 // Module version
+	uint16_t            version_build;                  // Module version
+	enum mod_type module_type;
+	uint32_t            init_settings_min_size;         // Minimum size of initialization settings (in bytes)
+	uint16_t            num_scheduling_capabilities;    // number scheduling capabilities supported by the module
+	uint16_t            num_pin_entries;                // Number of Pin (inputs + ouptuts)
```

Comment:
Can we make sure all comments are doxygen style and that we pack the data. The alignment looks off in places too.

Reply:
yes, thanks!